frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx

```diff
@@ -372,12 +391,16 @@ class ExpressionEditorTextfield extends React.Component<
     this.setState({ errorMessage });
 
     if (errorMessage) {
-      this.props.onError(errorMessage);
+      onError(errorMessage);
     } else {
-      const expression = this.compileExpression();
+      const compiledExpression = this.compileExpression();
+
+      if (compiledExpression) {
```

Comment:
I suspect we should add 
```tsx
onError({ message: t`Invalid expression` });
```
if `compiledExpression` is `null`, just like we do in `handleInputBlur`